src/modules/languages/rust.nix

```diff
@@ -31,10 +32,13 @@ in
 
   config = lib.mkMerge [
     (lib.mkIf cfg.enable {
-      packages = [
-        cfg.packages.rustc
-        cfg.packages.cargo
-      ];
+      packages = attrValues (getAttrs [ "rustc" "cargo" "rustfmt" "clippy" "rust-analyzer" ] cfg.packages);
+      pre-commit.tools.cargo = lib.mkForce cfg.packages.cargo;
+      pre-commit.tools.rustfmt = lib.mkForce cfg.packages.rustfmt;
+      pre-commit.tools.clippy = lib.mkForce cfg.packages.clippy;
+    })
+    (lib.mkIf (cfg.version == null) {
+      env.RUST_SRC_PATH = pkgs.rustPlatform.rustLibSrc;
```

Comment:
I wonder if this should be configurable. If you can specify a custom `rustc` package, then you would likely want to set this as well.

Reply:
Done